pkg/cmd/openshift-sdn-controller/interfaces.go

```diff
@@ -37,10 +42,24 @@ func newControllerContext(clientConfig *rest.Config) (*controllerContext, error)
 		osdnInformers:       osdninformer.NewSharedInformerFactory(osdnClient, defaultInformerResyncPeriod),
 	}
 
+	if strings.ToLower(platformType) == strings.ToLower(string(configv1.AWSPlatformType)) ||
+		strings.ToLower(platformType) == strings.ToLower(string(configv1.AzurePlatformType)) ||
+		strings.ToLower(platformType) == strings.ToLower(string(configv1.GCPPlatformType)) {
```

Comment:
Case-insensitivity is generally a bad idea because inevitably someone ends up doing a case-sensitive check somewhere. (This happened with `networkType` in the network config object.) Just require the caller to pass the correct string in.

Also, maybe have a `common.PlatformUsesCloudEgressIPs`?

Reply:
+1. This will need maintenance when we add other platforms such as Alibaba/IBM Cloud.